src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -1052,6 +1109,8 @@ private void WriteEndImplementation()
                         }
 
                         break;
+                    case WriterState.Primitive:
+                        break;

```

Comment:
Yep; WriteEnd for WriterState.Primitive is a no-op; just leave pop scope.

---
In reply to: [128098585](https://github.com/OData/odata.net/pull/894#discussion_r128098585) [](ancestors = 128098585)